gsrs-module-substance-example/src/test/java/example/substance/validation/StandardNameDuplicateValidatorTest.java

```diff
@@ -0,0 +1,127 @@
+package example.substance.validation;
+
+import gsrs.substances.tests.AbstractSubstanceJpaEntityTest;
+import ix.core.models.Keyword;
+import ix.core.validator.ValidationResponse;
+import ix.ginas.models.EmbeddedKeywordList;
+import ix.ginas.models.v1.Name;
+import ix.ginas.models.v1.Substance;
+import ix.ginas.utils.validation.validators.StandardNameDuplicateValidator;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import gsrs.cache.GsrsCache;
+import gsrs.module.substance.indexers.SubstanceDefinitionalHashIndexer;
+import gsrs.module.substance.repository.SubstanceRepository;
+import gsrs.springUtils.AutowireHelper;
+import gsrs.startertests.TestIndexValueMakerFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.security.test.context.support.WithMockUser;
+import java.io.File;
+import java.io.IOException;
+
+@WithMockUser(username = "admin", roles="Admin")
+@Slf4j
+public class StandardNameDuplicateValidatorTest extends AbstractSubstanceJpaEntityTest {
+
+@Autowired
+private SubstanceRepository substanceRepository;
+
+@Autowired
+private GsrsCache cache;
+
+@Autowired
+private TestIndexValueMakerFactory testIndexValueMakerFactory;
+
+
+@BeforeEach
+public void runSetup() throws IOException {
+        log.trace("runSetup");
+        AutowireHelper.getInstance().autowireAndProxy(substanceRepository);
+        SubstanceDefinitionalHashIndexer hashIndexer = new SubstanceDefinitionalHashIndexer();
+        AutowireHelper.getInstance().autowire(hashIndexer);
+        testIndexValueMakerFactory.addIndexValueMaker(hashIndexer);
+        // prevent validations from occurring multiple times
+        File dataFile = new ClassPathResource("testdumps/rep18.gsrs").getFile();
+        cache.clearCache();
+        loadGsrsFile(dataFile);
+        log.trace("loaded rep18 data file");
+}
+
+@Test
+public void testDuplicateInOtherRecord() {
+        // This test is not working as expected.
+
```

Comment:
In the validator there is a checkStdNameForDuplicateInOtherRecords methods with line 

            List<SubstanceRepository.SubstanceSummary> sr = substanceRepository.findByNames_StdNameIgnoreCase(stdName);

substanceRepository in this line is giving a NPE.  

I am not sure whether the line in runSetup in this test class is conveying the substance repository value outside the test context and inside the StandardNameDuplicateValidator. 

        AutowireHelper.getInstance().autowireAndProxy(substanceRepository);